### PR TITLE
CVE-2017-5715 AArch64

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/kernel/thread_a64.S [hunk 1]
```diff
@@ -507,6 +529,117 @@ el0_serror_a32:
 	b	el0_serror_a32
 	check_vector_size el0_serror_a32
 
+#if defined(CFG_CORE_WORKAROUND_SPECTRE_BP_SEC)
+	.macro apply_workaround
```
Comment (jforissier) [inline core/arch/arm/kernel/thread_a64.S]: `apply_workaround` is a bit too vague IMO. `s/apply_workaround/ic_iallu_isb/`? Or if you prefer `s/apply_workaround/icache_invalidate/`?

File: core/arch/arm/kernel/thread_a64.S [hunk 2]
```diff
@@ -379,31 +379,31 @@ END_FUNC thread_unwind_user_mode
 	.endm
 
 	.section .text.thread_vect_table
-	.align	11
+	.align	11, 0
```
Comment (jforissier) [inline core/arch/arm/kernel/thread_a64.S]: How could we make it more obvious that 0 is an invalid instruction? `#define INV_INSN 0` maybe?

Comment (jforissier) [reply core/arch/arm/kernel/thread_a64.S]: Commit subject: `s/arm32/arm64`

File: core/arch/arm/kernel/thread_a64.S (hunk 1 above)
Comment (jenswi-linaro) [reply core/arch/arm/kernel/thread_a64.S]: ARM-TF is mentioning a low overhead SMC interface for to invalidate the branch predictor, maybe that could be used later.
I guess `s/apply_workaround/invalidate_branch_predictor/` would be good enough.

File: core/arch/arm/kernel/thread_a64.S (hunk 2 above)
Comment (jenswi-linaro) [reply core/arch/arm/kernel/thread_a64.S]: I'll fix

**Comment by jenswi-linaro** [comment]
Comments addressed, tags applied.

File: core/arch/arm/kernel/thread_a64.S (hunk 1 above)
Comment (jforissier) [reply core/arch/arm/kernel/thread_a64.S]: Sounds good

**Comment by jenswi-linaro** [comment]
Rebased on master

**Pull request merged**